astropy/time/formats.py

```diff
@@ -731,17 +731,28 @@ def parse_string(self, timestr, subfmts):
             raise ValueError('Time {0} does not match {1} format'
                              .format(timestr, self.name))
 
+    @staticmethod
+    def _decode_if_needed(val, encoding='ascii'):
+        """Decode ``vall`` if it is a bytes instance."""
+        if type(result) is bytes:
+            result = result.decode(encoding)
+        return result
+
     def set_jds(self, val1, val2):
         """Parse the time strings contained in val1 and set jd1, jd2"""
         # Select subformats based on current self.in_subfmt
         subfmts = self._select_subfmts(self.in_subfmt)
 
+        # We cannot just use the iterator over val1, since that effectively
```

Comment:
This comment completely confuses me.  Is it a leftover from previous implementation?  Maybe below a comment to the effect that the `Column` unicode sandwich machinery doesn't work here because of how `np.nditer` works, hence the need to specifically decode each element in the case of a bytestring array `val.dtype='S'`.

(At least, that is my understanding of the situation..)

[EDIT] - re-reading the PR description it looks like this is a leftover from the multi-index idea.